File: Data Structures & Algorithms/maximum-product-subarray/submission-5.java
public class Solution {
    public int maxProduct(List<Integer> nums) {
        int res = nums.get(0);

        for (int i = 0; i < nums.size(); i++) {
            int cur = nums.get(i);
            res = Math.max(res, cur);
            for (int j = i + 1; j < nums.size(); j++) {
                cur *= nums.get(j);
                res = Math.max(res, cur);
            }
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-5.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int n = nums.Length, res = nums[0];
        for (int i = 0; i < n; i++) {
            int cur = 0;
            for (int j = i; j < n; j++) {
                cur += nums[j];
                res = Math.Max(res, cur);
            }
        }
        return res;
    }
}